streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java

```diff
@@ -691,19 +696,156 @@ protected StreamTask createStreamTask(TaskId id, Collection<TopicPartition> part
     }
 
     @Test
-    public void testInjectClients() {
-        TopologyBuilder builder = new TopologyBuilder().setApplicationId("X");
-        StreamsConfig config = new StreamsConfig(configProps());
-        MockClientSupplier clientSupplier = new MockClientSupplier();
-        StreamThread thread = new StreamThread(builder, config, clientSupplier, applicationId,
-                                               clientId, processId, new Metrics(), new MockTime(), new StreamsMetadataState(builder, StreamsMetadataState.UNKNOWN_HOST),
-                                               0);
-        assertSame(clientSupplier.producer, thread.producer);
+    public void shouldUseClientSupplierToInjectClients() {
```

Comment:
Ah -- forgot that I did the split already -- haha -- too much stuff going on atm... Will do the renaming :)